src/components/InjectedComponents/PostDialog.tsx

```diff
@@ -342,30 +345,25 @@ export function PostDialog(props: PostDialogProps) {
             setPostBoxContent(makeTypedMessage(''))
             setCurrentShareTarget([])
             getActivatedUI().typedMessageMetadata.value = new Map()
-        }, []),
+        }, [setOpen]),
     )
     const onFinishButtonClicked = useCallback(() => {
         onRequestPost(onlyMyself ? [currentIdentity!] : currentShareTarget, postBoxContent)
         onRequestReset()
     }, [currentIdentity, currentShareTarget, onRequestPost, onRequestReset, onlyMyself, postBoxContent])
     const onCloseButtonClicked = useCallback(() => {
         setOpen(false)
-    }, [])
+    }, [setOpen])
     //#endregion
     //#region My Identity
     const identities = useMyIdentities()
-    const [open, setOpen] = useState(false)
     useEffect(() => {
-        const onChange = ({ reason, open }: CompositionEvent) => {
+        return MessageCenter.on('compositionUpdated', ({ reason, open }: CompositionEvent) => {
```

Comment:
Should we remove the event listener after component destroyed?

Reply:
the `on` function returns a function that will remove the listener.